### DISCUSSION TITLE
Build warnings

### DISCUSSION THREAD

**Discussion by 0323pin** [sxyazi/yazi#518] [1. Q&A]
Hi,

NetBSD package maintainer here.
For some reason, I haven't looked further into it, `mlua` doesn't build on NetBSD when the _vendored_ feature is enabled. So, I've patched it away and make it use the local copy of `LuaJIT2`.

Given this, I would still like to let you know about the following build warnings.

```
[...]
   Compiling nom v7.1.3
warning: glob import doesn't reexport anything because no candidate is public enough
 --> yazi-config/src/keymap/mod.rs:7:9
  |
7 | pub use exec::*;
  |         ^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: glob import doesn't reexport anything because no candidate is public enough
 --> yazi-config/src/plugin/mod.rs:5:9
  |
5 | pub use exec::*;
  |         ^^^^^^^

   Compiling image v0.24.7
[...]
   Compiling yazi-plugin v0.2.0 (/usr/pkgsrc/wip/yazi/work/yazi-0.2.0/yazi-plugin)
warning: unused import: `bindings::*`
  --> yazi-plugin/src/bindings/mod.rs:11:9
   |
11 | pub use bindings::*;
   |         ^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

   Compiling kqueue v1.0.8
   Compiling trash v3.2.0
warning: unused variable: `name`
  --> yazi-plugin/src/utils/plugin.rs:10:10
   |
10 |                 |_, (name, calls, args): (String, usize, Variadic<Value>)| async move { Ok(()) },
   |                      ^^^^ help: if this is intentional, prefix it with an underscore: `_name`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `calls`
  --> yazi-plugin/src/utils/plugin.rs:10:16
   |
10 |                 |_, (name, calls, args): (String, usize, Variadic<Value>)| async move { Ok(()) },
   |                            ^^^^^ help: if this is intentional, prefix it with an underscore: `_calls`

warning: unused variable: `args`
  --> yazi-plugin/src/utils/plugin.rs:10:23
   |
10 |                 |_, (name, calls, args): (String, usize, Variadic<Value>)| async move { Ok(()) },
   |                                   ^^^^ help: if this is intentional, prefix it with an underscore: `_args`

warning: field `stdin` is never read
  --> yazi-plugin/src/process/child.rs:10:2
   |
8  | pub struct Child {
   |            ----- field in this struct
9  |     inner:  tokio::process::Child,
10 |     stdin:  Option<ChildStdin>,
   |     ^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

   Compiling addr2line v0.21.0
[...]
warning: `yazi-plugin` (lib) generated 5 warnings (run `cargo fix --lib -p yazi-plugin` to apply 4 suggestions)
   Compiling yazi-fm v0.2.0 (/usr/pkgsrc/wip/yazi/work/yazi-0.2.0/yazi-fm)
warning: unused import: `root::*`
  --> yazi-fm/src/main.rs:24:5
   |
24 | use root::*;
   |     ^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: `yazi-fm` (bin "yazi") generated 1 warning (run `cargo fix --bin "yazi"` to apply 1 suggestion)
    Finished release [optimized] target(s) in 17m 29s
```

System is NetBSD dev. version 10.99.10 using Rust-1.75.0
Regards,

**Comment by sxyazi** [comment]
Yeah those are part of the plugin system that's still incomplete. I left them there to remind me to finish it.

Yazi uses lua5.4 for now, no longer needing LuaJIT. Does it still need to apply patches?

**Comment by 0323pin** [comment]
> Yazi uses lua5.4 for now, no longer needing LuaJIT. Does it still need to apply patches?

I would need to check this. The problem used to be the _vendored_ feature of `mlua`, this is not `yazi` specific, it happens with any package that wants to build `mlua` with that feature on.
But, I'll check what happens if I disabled LuaJIT and make it depend on Lua-5.4

Back to you when I've tested.

**Comment by 0323pin** [comment]
@sxyazi Indeed, it builds without patches and LuaJIT if I link it against lua-5.4

I'll update our package to 0.2.1 and remove the patch and LuaJIT dependency.

**Reply by sxyazi** [reply]
Thank you!

**Reply by 0323pin** [reply]
https://mail-index.netbsd.org/pkgsrc-changes/2024/01/16/msg291064.html

**Comment by 0323pin** [comment]
If you don't mind, I have another minor question.

I use `tuifeed` to get RSS feeds when a new version of the packages I maintain are released. For `yazi` the feed URL is https://github.com/sxyazi/yazi/tags.atom and/or https://github.com/sxyazi/yazi/releases.atom but, the latest version on both is still 0.1.5

Do you know why the last two releases don't exist on RSS? 

**EDIT:** Actually, it gets the dates right but, not the version 🤔 

![2024-01-17-074441_1366x768_scrot](https://github.com/sxyazi/yazi/assets/90570748/1bc25661-b600-4065-aa02-ac67e7402352)

**Comment by sxyazi** [comment]
![1](https://github.com/sxyazi/yazi/assets/17523360/1388d6b4-f3e1-4e51-9730-6c6d1f2bcb66)

It's at the end, possibly because I modified the release before v0.2.0 after publishing - removed the mistakenly checked "Set as a pre-release"

**Reply by 0323pin** [reply]
🤦‍♂️

**Comment by sxyazi** [comment]
warnings are cleared up now.
